src/frontend/js/widgets/UserLogin/index.tsx

```diff
@@ -72,11 +74,51 @@ const UserLogin = ({ profileUrls = {} }: UserLoginProps) => {
    * - a user object when the user is logged in.
    */
   const { user, destroy, login, register } = useSession();
+  const {
+    item: joanieUser,
+    states: { isFetched: isAbilitiesFetched },
+  } = useUser({ enabled: Boolean(user) });
   const intl = useIntl();
 
+  const unlogged = user === null;
+  const logged = !unlogged && user !== undefined;
+  const isReady = unlogged || (logged && isAbilitiesFetched);
+
+  const userMenuItems = useMemo(() => {
+    if (isReady) {
+      const unauthorizedUrlFilter = (urlItem: UserMenuItem) => {
+        if (urlItem.key === 'dashboard_teacher') {
+          if (isAbilitiesFetched) {
+            return (
+              joanieUser.abilities[JoanieAbilities.HAS_ORGANIZATION_ACCESS] ||
+              joanieUser.abilities[JoanieAbilities.HAS_COURSE_ACCESS]
+            );
+          }
```

Comment:
I'm thinking that this should move into a `Abitility` object that would be use as: 
```ts
Abilities.can(JoanieUser, Abilities.ACCESS_TEACHER_DASHBOARD)
```